implement depth bound extension (#62)
shapesecurity/superpack-js: README.md

let reconstructedValue = decode(payload);
```

## Extensions

`encode` and `decode` take a second argument which is an options bag. It allows specifying extensions as follows:

```js
encode(data, {
  extensions: {
    [extensionPoint]: {
      detector: value => false,
      serialiser: value => serialize(value),
      deserialiser: serializedValue => deserialise(serializedValue),
    }
  }
});
````

where `extensionPoint` is a number. The decoder must have the same extensions at the same extension points.

> ## Depth bound extension

Reviewer: ```suggestion
### Depth bound extension
```

> This implementation includes a buit-in extension which allows bounding the depth of object / array nesting allowed. Once reached, a sigil value is emitted instead. It is used as

Reviewer: ```suggestion
This implementation includes a buit-in extension which allows bounding the depth of object / array nesting. Once reached, a sentinel value is emitted instead. It is used as
```

```js

> import {encode, depthBoundExtension} from "superpack";

Reviewer: ```suggestion
import {encode, depthBoundExtension, depthBoundReached} from "superpack";
```

Showing use of `depthBoundReached` would be nice.
